google/cloud/firestore_v1/base_collection.py

```diff
@@ -455,6 +457,13 @@ def _auto_id() -> str:
         str: A 20 character string composed of digits, uppercase and
         lowercase and letters.
     """
+    if sys.version_info < (3, 7):
+        # TODO: remove when 3.6 support is discontinued.
+        # On python 3.6, random will provide the same results when forked. Reseed using a
+        # uuid4 on each iteration to avoid collisions.
+        seed = str(uuid.uuid4())
+        random.seed(seed)
```

Comment:
`uuid.uuid4` itself uses `os.urandom` module as its source of entropy, which is exactly what `random.seed` does if the `seed` argument is `None`, this would be simpler as:

```suggestion
        random.seed()
```

(and remove the import and use of `uuid`).

Reply:
I tried to reproduce a collision with just `random.seed()` and did not, but I have conerns as the docs state this will use system time as a seed, which seems to me there could be a collision.

Running the below code:

```
if pid == 0:
    r = random.random()
    id = uuid.uuid4()
    reference = client.collection("Test").document()
    print("Child: random.random:", r, "uuid4:", id, "documentId:", reference.id)
else:
    r = random.random()
    id = uuid.uuid4()
    reference = client.collection("Test").document()
    print("Parent: random.random:", r, "uuid4:", id, "documentId:", reference.id)
```

```
Parent: random.random: 0.28313230731702665 uuid4: 738ffed5-c52b-44cf-ba9b-1118ad9921c0
Child: random.random: 0.28313230731702665 uuid4: 0ae5a7ae-4345-4154-8c9e-c988bda1cd7e
```

Across forks `random.random` is the same, but `uuid4` is unique. The source in cpython for uuid4 is identical in 3.6 and 3.9 branches, but random.py has seen a reasonable amount of change, possibly in the sha512 c module also.

It does seem that `random.seed` improves things, I am just not sure it won't ever collide.

Reply:
`random.seed` prefers to use the system `urandom` value, but falls back to system time + PID if it the `urandom` bits are exhausted, or the system doesn't support `urandom`.  This behavior is consistent across 3.6 - 3.9.

Reply:
Seems like we can simplify then. I am not going to spend too much time seeing as this fix will be redundant in 6 months as 3.7+ behave correctly. There seems to be a small change between 3.6 and 3.7 with a lack of randomness on fork.

Reply:
LGTM